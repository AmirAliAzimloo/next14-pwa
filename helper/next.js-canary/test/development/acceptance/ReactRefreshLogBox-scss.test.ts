/* eslint-env jest */
import { sandbox } from 'development-sandbox'
import { FileRef, nextTestSetup } from 'e2e-utils'
import path from 'path'

// TODO: figure out why snapshots mismatch on GitHub actions
// specifically but work in docker and locally
describe.skip('ReactRefreshLogBox', () => {
  const { next } = nextTestSetup({
    files: new FileRef(path.join(__dirname, 'fixtures', 'default-template')),
    skipStart: true,
    dependencies: {
      sass: 'latest',
    },
  })

  test('scss syntax errors', async () => {
    const { session, cleanup } = await sandbox(next)

    await session.write('index.module.scss', `.button { font-size: 5px; }`)
    await session.patch(
      'index.js',
      `
        import './index.module.scss';
        export default () => {
          return (
            <div>
              <p>lol</p>
            </div>
          )
        }
      `
    )

    expect(await session.hasRedbox(false)).toBe(false)

    // Syntax error
    await session.patch('index.module.scss', `.button { font-size: :5px; }`)
    expect(await session.hasRedbox(true)).toBe(true)
    const source = await session.getRedboxSource()
    expect(source).toMatchSnapshot()

    // Not local error
    await session.patch('index.module.scss', `button { font-size: 5px; }`)
    expect(await session.hasRedbox(true)).toBe(true)
    const source2 = await session.getRedboxSource()
    expect(source2).toMatchSnapshot()

    await cleanup()
  })
})
