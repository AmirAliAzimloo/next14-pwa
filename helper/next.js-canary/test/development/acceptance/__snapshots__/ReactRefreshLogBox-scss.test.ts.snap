// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactRefreshLogBox scss syntax errors 1`] = `
"./index.module.scss
SassError: Expected expression.
  ╷
1 │ .button { font-size: :5px; }
  │                      ^
  ╵
  index.module.scss 1:22  root stylesheet"
`;

exports[`ReactRefreshLogBox scss syntax errors 2`] = `
"./index.module.scss:1:1
Syntax error: Selector \\"button\\" is not pure (pure selectors must contain at least one local class or id)

> 1 | button { font-size: 5px; }
    |                          ^"
`;
